### DISCUSSION TITLE
List of shape types

### DISCUSSION THREAD

**Discussion by mickwadev** [ShapeCrawler/ShapeCrawler#213] [Q&A] [answered]
Hello :)

In example there is:
```
// Get picture shape
IPicture picture = presentation.Slides[0].Shapes.OfType<IPicture>().First();
```
Is there a list of interfaces that can be used here in place of IPicture for searching texts, charts, etc?

Best regards,
Michał

**Comment by AdamShakhabov** [comment, marked as answer]
Good question :) Looks like it is time for documenting such things because these entities are stable.

So, `IShape` item can be one of the following derived interface:

![image](https://user-images.githubusercontent.com/5969304/152397804-8a284094-ca82-4dd0-a6c8-6a06d81478fe.png)
![image](https://user-images.githubusercontent.com/5969304/152397735-dbc3b73c-0cd7-4d59-af18-56ef37487247.png)
